### DISCUSSION TITLE
Enum constants can have properties too

### DISCUSSION THREAD

**Discussion by JordanLongstaff** [LemonAppDev/konsist#1596] [Request a Feature] [answered]
### Tell us more.

Here's an example enum class.

```
enum class PrivateNetworkType {
    TWENTY_FOUR_BIT_BLOCK {
        // 10.x.x.x
        override val broadcastAddress: String get() = TWENTY_FOUR_BIT_BROADCAST
        override val constraints: Array<ByteConstraint> get() = TWENTY_FOUR_BIT_CONSTRAINTS
    },
    TWENTY_BIT_BLOCK {
        // 172.16.x.x - 172.31.x.x
        override val broadcastAddress: String get() = TWENTY_BIT_BROADCAST
        override val constraints: Array<ByteConstraint> get() = TWENTY_BIT_CONSTRAINTS
    },
    SIXTEEN_BIT_BLOCK {
        // 192.168.x.x
        override val broadcastAddress: String get() = SIXTEEN_BIT_BROADCAST
        override val constraints: Array<ByteConstraint> get() = SIXTEEN_BIT_CONSTRAINTS
    };

    abstract val broadcastAddress: String
    internal abstract val constraints: Array<ByteConstraint>
}
```

These enum constants have properties. These properties have getters. In this case, they are overrides of abstract properties. Enum classes can also have constructors with property arguments. None of these conventions are any different from properties of a class.
However, Konsist does not acknowledge the possibility of enum constants having properties. Instead, they are "variables" which is both incorrect and obscures a lot of useful properties of these declarations.

**Comment by nataliapeterwas** [comment]
Thanks for reporting this issue. I've created a ticket to fix it: https://lemonappdev.atlassian.net/browse/KON-695

**Comment by nataliapeterwas** [comment, marked as answer]
Will be added in the next release: [PR](https://github.com/LemonAppDev/konsist/pull/1618)
